source/objects/simulationClass.m

```diff
@@ -28,45 +28,45 @@
+        nonlinearDt (1,:) double {mustBeScalarOrEmpty}  = []                % (`float`) Sample time to calculate nonlinear forces. Default = ``dt``
+        paraview (1,1) struct                           = struct(...        % (`structure`) Defines the Paraview visualization.
+            'option',                                   0,...               % 
+            'startTime',                                0, ...              %     
+            'endTime',                                  100, ...            % 
+            'dt',                                       0.1, ...            % 
+            'path',                                     'vtk')              % (`structure`) Defines the Paraview visualization. ``option`` (`integer`) Flag for paraview visualization, and writing vtp files, Options: 0 (off) , 1 (on). Default = ``0``. ``startTime`` (`float`) Start time for the vtk file of Paraview. Default = ``0``. ``endTime`` (`float`) End time for the vtk file of Paraview. Default = ``100``.  ``dt`` (`float`) Timestep for Paraview. Default = ``0.1``. ``path`` (`char array`) Path of the folder for Paraview vtk files. Default = ``'vtk'``.      
+        pressure (1,1) double                           = 0                 % (`integer`) Flag to save pressure distribution, Options: 0 (off), 1 (on). Default = ``0``
+        rampTime (1,1) double                           = 100               % (`float`) Ramp time for wave forcing. Default = ``100`` s        
+        rateTransition (1,:) char                       = 'on'              % (`char array`) Flag for automatically handling rate transition for data transfer, Opyions: 'on', 'off'. Default = ``'on'``
+        reloadH5Data (1,1) double                       = 0                 % (`integer`) Flag to re-load hydro data from h5 file between runs, Options: 0 (off), 1 (on). Default = ``0``
+        rho (1,1) double {mustBePositive}               = 1000              % (`float`) Density of water. Default = ``1000`` kg/m^3
+        saveStructure (1,1) double                      = 0                 % (`integer`) Flag to save results as a MATLAB structure, Options: 0 (off), 1 (on). Default = ``0``
+        saveText (1,1) double                           = 0                 % (`integer`) Flag to save results as ASCII files, Options: 0 (off), 1 (on). Default = ``0``
+        saveWorkspace (1,1) double                      = 1                 % (`integer`) Flag to save .mat file for each run, Options: 0 (off), 1 (on). Default = ``1``
+        simMechanicsFile (1,:) char                     = 'NOT DEFINED'     % (`char array`) Simulink/SimMechanics model file. Default = ``'NOT DEFINED'``
+        solver (1,:) char                               = 'ode4'            % (`char array`) PDE solver used by the Simulink/SimMechanics simulation. Any continuous solver in Simulink possible. Recommended to use 'ode4, 'ode45' for WEC-Sim. Default = ``'ode4'``

```

Comment:
@jtgrasb I have been spot checking the "checks", but when I write `simu.solver = 1` the checkinputs does not seem to catch this and the error that pops up basically says this is not a valid solver. Therefore, in my opinion it does not seem to be catching this case.

Reply:
@nathanmtom It seems the char specification tries to convert the number to a char array, which leads to just an empty char. I am adding a check in simu.checkInputs to ensure the solver is one of Simulink's solvers.